### ISSUE TITLE
Removal of PYTHONHOME from Python package causes build errors with ecflow

### ISSUE THREAD

**Issue by climbfuji** [LLNL/spack#34028]
### Steps to reproduce

The ecflow package apparently requires PYTHONHOME to be set to the Python that is used by spack, for example when compiling boost. PYTHONHOME was removed from the Python `package.py` recently. In our macOS CI tests, this results in an error that boost builds correctly against the Python we want it to use (3.9.12 from homebrew), but ecflow picks the macOS system Python (3.11) and then complains that boost is missing the Python libraries for that version 3.11.

A workaround is to put the recently deleted PYTHONHOME back into package.py (see https://github.com/NOAA-EMC/spack/pull/198/commits/2bd85bbe9c6a26225b70b6d50f850e2cad524054). Another workaround could be to define PYTHONHOME in the ecflow package itself, but maybe there are other packages that also need PYTHONHOME, which nobody has built since the change was made? 

### Error message

See https://github.com/NOAA-EMC/spack-stack/actions/runs/3498920101/jobs/5859878661 for full build log. Error:
```
[+] /Users/runner/work/spack-stack/spack-stack/envs/default/install/apple-clang/13.0.0/py-pyhdf-0.10.4-qw6mt7j
==> Installing ecflow-5.8.3-7ouale3bwwqeirejcpssszzwzjyyltcs
==> No binary for ecflow-5.8.3-7ouale3bwwqeirejcpssszzwzjyyltcs found: installing from source
==> Fetching https://confluence.ecmwf.int/download/attachments/8650755/ecFlow-5.8.3-Source.tar.gz
==> No patches needed for ecflow
==> ecflow: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    '/usr/local/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/Users/runner/work/spack-stack/spack-stack/envs/default/install/apple-clang/13.0.0/ecflow-5.8.3-7ouale3' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DBUILD_TESTING:BOOL=OFF' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_FIND_FRAMEWORK:STRING=LAST' '-DCMAKE_FIND_APPBUNDLE:STRING=LAST' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/Users/runner/work/spack-stack/spack-stack/envs/default/install/apple-clang/13.0.0/ecflow-5.8.3-7ouale3/lib;/Users/runner/work/spack-stack/spack-stack/envs/default/install/apple-clang/13.0.0/ecflow-5.8.3-7ouale3/lib64;/Users/runner/work/spack-stack/spack-stack/envs/default/install/apple-clang/13.0.0/boost-1.78.0-4jfk5im/lib;/Users/runner/work/spack-stack/spack-stack/envs/default/install/apple-clang/13.0.0/openssl-1.1.1s-kifks5y/lib;/Users/runner/work/spack-stack/spack-stack/envs/default/install/apple-clang/13.0.0/zlib-1.2.13-rlzphst/lib' '-DCMAKE_PREFIX_PATH:STRING=/Users/runner/work/spack-stack/spack-stack/envs/default/install/apple-clang/13.0.0/py-numpy-1.22.3-65fdcy2;/Users/runner/work/spack-stack/spack-stack/envs/default/install/apple-clang/13.0.0/py-setuptools-59.4.0-tqtc43h;/Users/runner/work/spack-stack/spack-stack/envs/default/install/apple-clang/13.0.0/py-pip-22.2.2-pc4oqms;/Users/runner/work/spack-stack/spack-stack/envs/default/install/apple-clang/13.0.0/openssl-1.1.1s-kifks5y;/Users/runner/work/spack-stack/spack-stack/envs/default/install/apple-clang/13.0.0/zlib-1.2.13-rlzphst;/Users/runner/work/spack-stack/spack-stack/envs/default/install/apple-clang/13.0.0/boost-1.78.0-4jfk5im' '-DBoost_PYTHON_LIBRARY_RELEASE=/Users/runner/work/spack-stack/spack-stack/envs/default/install/apple-clang/13.0.0/boost-1.78.0-4jfk5im/lib' '-DENABLE_STATIC_BOOST_LIBS=OFF' '-DENABLE_SSL=OFF' '-DENABLE_UI=OFF' '-DENABLE_GUI=OFF' '/Users/runner/work/spack-stack/spack-stack/cache/build_stage/spack-stage-ecflow-5.8.3-7ouale3bwwqeirejcpssszzwzjyyltcs/spack-src'

1 error found in build log:
     109    --   Python3_VERSION                   : 3.11.0
     110    --   Python3_VERSION_MAJOR             : 3
     111    --   Python3_VERSION_MINOR             : 11
     112    --   Python3_VERSION_PATCH             : 0
     113    -- Could NOT find Boost (missing: python311) (found suitable versio
            n "1.78.0", minimum required is "1.66.0")
     114    --   Boost python3 libraries *NOT* found
  >> 115    CMake Error at cmake/ecbuild_log.cmake:172 (message):
     116      ERROR - ecflow python extension is enabled, but python2/python3
     117      libraries or python boost libraries not found
     118    Call Stack (most recent call first):
     119      Pyext/CMakeLists.txt:180 (ecbuild_error)
     120    
     121    

See build log for details:
  /Users/runner/work/spack-stack/spack-stack/cache/build_stage/spack-stage-ecflow-5.8.3-7ouale3bwwqeirejcpssszzwzjyyltcs/spack-build-out.txt
```

### Information on your system

```
* **Spack:** 0.20.0.dev0 (2bd85bbe9c6a26225b70b6d50f850e2cad524054)
* **Python:** 3.9.13
* **Platform:** - Github actions using macos-11 CI image
* **Concretizer:** clingo
```

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by adamjstewart** [comment]
@alalazo

**Comment by alalazo** [comment]
Thanks for the ping @adamjstewart I'll have a look at this asap.

**Comment by alalazo** [comment]
I see you have an updated version of `ecflow`! Would it be possible to upstream it? With the one in Spack the issues are old bugs due to incompatibility of Boost 1.69 and Python 3.10

**Comment by alalazo** [comment]
The thing I'd try, once I have the updated version of the recipe you're working on, is to bump the requirement on CMake to v3.16+ and set [Python_EXECUTABLE](https://cmake.org/cmake/help/latest/module/FindPython.html#artifacts-specification) and other similar variables as needed to help CMake pick the correct interpreter.

**Comment by climbfuji** [comment]
> The thing I'd try, once I have the updated version of the recipe you're working on, is to bump the requirement on CMake to v3.16+ and set [Python_EXECUTABLE](https://cmake.org/cmake/help/latest/module/FindPython.html#artifacts-specification) and other similar variables as needed to help CMake pick the correct interpreter.

Yeah, I was looking at the ECMWF atlas package (which we have as ecmwf-atlas in our fork, not yet in the spack repo) and for that one I am already setting `Python_EXECUTABLE`. See https://github.com/NOAA-EMC/spack/blob/d8bfe52ebbddf485c4455c64aaf8a79e8abfb3cc/var/spack/repos/jcsda-emc/packages/ecmwf-atlas/package.py#L69.

**Comment by climbfuji** [comment]
> I see you have an updated version of `ecflow`! Would it be possible to upstream it? With the one in Spack the issues are old bugs due to incompatibility of Boost 1.69 and Python 3.10

Yes, let me do this asap, and then we can try without PYTHONHOME in the Python pacakge, but with Python_EXECUTABLE in ecflow.

**Comment by alalazo** [comment]
For what is worth I can build the old version we have in Spack with some fixing to `Pyext/CMakeLists.txt`. Not sure what I did is still relevant for the next major version of `ecflow`.

**Comment by climbfuji** [comment]
https://github.com/spack/spack/pull/34038

**Comment by alalazo** [comment]
Closed by #34043
